packages/sanity/src/core/releases/tool/components/releaseCTAButtons/ReleaseRevertButton/useDocumentRevertStates.ts

```diff
@@ -0,0 +1,135 @@
+import {type SanityDocument} from '@sanity/types'
+import {useCallback, useEffect, useMemo, useRef} from 'react'
+import {useObservable} from 'react-rx'
+import {catchError, filter, forkJoin, from, map, type Observable, of, switchMap} from 'rxjs'
+
+import {useClient} from '../../../../../hooks/useClient'
+import {getTransactionsLogs} from '../../../../../store/translog/getTransactionLogs'
+import {API_VERSION} from '../../../../../tasks/constants'
+import {type DocumentInRelease} from '../../../detail/useBundleDocuments'
+
+export type RevertDocument = SanityDocument & {
+  _system?: {
+    delete: boolean
+  }
+}
+
+type RevertDocuments = RevertDocument[]
+
+type DocumentRevertStates = RevertDocuments | null | undefined
+
+export const useDocumentRevertStates = (documents: DocumentInRelease[]) => {
```

Comment:
Could I have some TSDoc with a short explanation of what this does? Struggled a little bit with unpacking the implementation 😅

Reply:
On a second readthrough, I'm wondering if there is a reason this has to be a hook – seems like a better candidate for a regular async function to me, but maybe I'm missing something?